core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -89,7 +92,7 @@ class HadoopMapReduceCommitProtocol(
    * The staging directory of this write job. Spark uses it to deal with files with absolute output
    * path, or writing data into partitioned directory with dynamicPartitionOverwrite=true.
    */
-  private def stagingDir = new Path(path, ".spark-staging-" + jobId)
+  private def stagingDir = new Path(stagingPath.getOrElse(path))
```

Comment:
AFAIK, this staging dir is used to deal with files with absolute output path, or writing data into partitioned directory with dynamicPartitionOverwrite=true.

But in this PR, it change the behavior for absolute output path.

Reply:
You are right, stagingDir cannot be used here, as it might get deleted during abortJob.